service/history/historyEngine.go

```diff
@@ -1528,6 +1560,17 @@ func (e *historyEngineImpl) SignalWorkflowExecution(signalRequest *h.SignalWorkf
 				return &workflow.EntityNotExistsError{Message: "Workflow execution already completed."}
 			}
 
+			// deduplicate by request id for signal decision
+			if request.RequestId != nil {
```

Comment:
This can be simplified if you use the generated getter.

Reply:
Changed, good to know